Framework/Crystal/src/IntegratePeakTimeSlices.cpp

```diff
@@ -259,18 +259,14 @@ void IntegratePeakTimeSlices::exec() {
   double Col0 = lastCol;
   string spec_idList;
 
-  // For quickly looking up workspace index from det id
-  m_wi_to_detid_map = inpWkSpace->getDetectorIDToWorkspaceIndexMap();
+  const auto &detectorInfo = inpWkSpace->detectorInfo();
 
   TableWorkspace_sptr TabWS = boost::make_shared<TableWorkspace>(0);
 
   //----------------------------- get Peak extents
   //------------------------------
   try {
-
-    // Find the workspace index for this detector ID
-    detid2index_map::const_iterator it = m_wi_to_detid_map.find(detID);
-    size_t wsIndx = (it->second);
+    const size_t wsIndx = detectorInfo.indexOf(detID);
```

Comment:
This is not what you need. The index returned by `indexOf` is an index for the detector, but you need an index for the detector group (i.e., the index of spectrum which contains the detector). I think you need to keep the old way of obtaining the index.